test/src/unittests/tonal/test_pitchsaliencefunction.py

```diff
@@ -0,0 +1,223 @@
+        self.assertConfigureFails(PitchSalienceFunction(), {'magnitudeCompression': -1})
+        self.assertConfigureFails(PitchSalienceFunction(), {'magnitudeCompression': 2})        
+        self.assertConfigureFails(PitchSalienceFunction(), {'magnitudeThreshold': -1})
+        self.assertConfigureFails(PitchSalienceFunction(), {'numberHarmonics': 0})
+        self.assertConfigureFails(PitchSalienceFunction(), {'numberHarmonics': -1})        
+        self.assertConfigureFails(PitchSalienceFunction(), {'referenceFrequency': -1})
+
+    def testEmpty(self): 
+        self.assertEqualVector(PitchSalienceFunction()([], []), zeros(600))
+
+    # Provide a single input peak with a unit magnitude at the reference frequency, and 
+    # validate that the output salience function has only one non-zero element at the first bin.
+    def testSinglePeak(self):
+        freq_speaks = [55] # length 1
+        mag_speaks = [1] # length 4
+        calculatedPitchSalience = PitchSalienceFunction()(freq_speaks,mag_speaks)    
+        self.assertEqual(calculatedPitchSalience[0],1)
+
+    def test3Peaks(self):
+        freq_speaks = [55, 100, 340] # length 1
+        mag_speaks = [1, 1, 1] # length 4
+        # For a single frequency 55 Hz with unitary  amplitude the 10 non zero salience function values are the following
+        calculatedPitchSalience = PitchSalienceFunction()(freq_speaks,mag_speaks)    
+        self.assertAlmostEqual(calculatedPitchSalience[0], 1.16384, 3)
+        self.assertEqual(len(calculatedPitchSalience), 600)        
+
+    def test3PeaksHw0(self):
+        freq_speaks = [55, 100, 340] # length 1
+        mag_speaks = [1, 1, 1] # length 4
+        # For a single frequency 55 Hz with unitary  amplitude the 10 non zero salience function values are the following
+
+        calculatedPitchSalience = PitchSalienceFunction(harmonicWeight=0)(freq_speaks, mag_speaks)    
+        self.assertAlmostEqual(calculatedPitchSalience[0], 1.16384, 3)
+        self.assertEqual(len(calculatedPitchSalience), 600)
+
+    def test3PeaksHw1(self):
+        freq_speaks = [55, 100, 340] # length 1
+        mag_speaks = [1, 1, 1] # length 4
+        # For a single frequency 55 Hz with unitary  amplitude the 10 non zero salience function values are the following
+        calculatedPitchSalience = PitchSalienceFunction(harmonicWeight=1)(freq_speaks, mag_speaks) 
+        self.assertAlmostEqual(calculatedPitchSalience[0], 1.5, 3)
+
+    def testDifferentPeaks(self):
+        freq_speaks = [55, 85] # length 1
+        mag_speaks = [1, 1] # length 4
+        calculatedPitchSalience1 = PitchSalienceFunction()(freq_speaks,mag_speaks)
+              
+        mag_speaks = [0.5, 2] # length 4
+        calculatedPitchSalience2 = PitchSalienceFunction()(freq_speaks,mag_speaks)
+        self.assertAlmostEqual(calculatedPitchSalience1[0], 0.5, 3)        
+        self.assertAlmostEqual(calculatedPitchSalience2[0], 1.5, 3)
```

Comment:
Here the rest of the values in the output vector aren't covered by the tests as well.